src/database/keyvalue.rs

```diff
@@ -166,16 +166,9 @@ macro_rules! impl_batch_operations {
         fn del_last_index(&mut self, keychain: KeychainKind) -> Result<Option<u32>, Error> {
             let key = MapKey::LastIndex(keychain).as_map_key();
             let res = self.remove(key);
-            let res = $process_delete!(res);
-
-            match res {
-                None => Ok(None),
-                Some(b) => {
-                    let array: [u8; 4] = b.as_ref().try_into().map_err(|_| Error::InvalidU32Bytes(b.to_vec()))?;
-                    let val = u32::from_be_bytes(array);
-                    Ok(Some(val))
-                }
-            }
+            $process_delete!(res)
+            .map(ivec_to_u32)
+            .transpose()
```

Comment:
I just noticed that there wasn't a `.transpose()` in the original code, did you mean to add it here?

Reply:
Yes. I believed that's the correct behaviour. it just replace the match expression.
`None ->Ok(None)`, `Some(Ok(_))` -> `Ok(Some(_))` and `Ok(Err(_))` -> `Err(_)`
https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose

Reply:
Thanks for the explanation I love learning cool Rust like this!